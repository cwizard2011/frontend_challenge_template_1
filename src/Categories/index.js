import React from 'react'

const Category = ({ setCategory, category }) => {
  return (
    <div style={{
      position: 'absolute',
      marginLeft: 10,
      boxShadow: '0.5rem 0.5rem 3rem rgba(0,0,0,0.2)',
      backgroundColor: '#FAFAFA',
      paddingLeft: 30,
      paddingRight: 30,
      paddingTop: 20,
      marginTop: 30,
      borderRadius: 5,
      paddingBottom: 10
    }}>
      <h4>Categories</h4>
      <p
        onClick={() => setCategory('french', '1')}
        style={{
          backgroundColor: category === 'french' ? 'red' : '',
          textAlign: 'center',
          paddingTop: category === 'french' ? 5 : '',
          paddingBottom: category === 'french' ? 5 : '',
          borderRadius: category === 'french' ? 5 : '',
          color: category === 'french' ? 'white' : 'black',
          cursor: 'pointer'
        }}
      >French</p>
      <p
        onClick={() => setCategory('italian', '2')}
        style={{
          backgroundColor: category === 'italian' ? 'red' : '',
          textAlign: 'center',
          paddingTop: category === 'italian' ? 5 : '',
          paddingBottom: category === 'italian' ? 5 : '',
          borderRadius: category === 'italian' ? 5 : '',
          color: category === 'italian' ? 'white' : 'black',
          cursor: 'pointer'
        }}
      >Italian</p>
      <p
        onClick={() => setCategory('irish', '3')}
        style={{
          backgroundColor: category === 'irish' ? 'red' : '',
          textAlign: 'center',
          paddingTop: category === 'irish' ? 5 : '',
          paddingBottom: category === 'irish' ? 5 : '',
          borderRadius: category === 'irish' ? 5 : '',
          color: category === 'irish' ? 'white' : 'black',
          cursor: 'pointer'
        }}
        >Irish</p>
      <p
        onClick={() => setCategory('animal', '4')}
        style={{
          backgroundColor: category === 'animal' ? 'red' : '',
          textAlign: 'center',
          paddingTop: category === 'animal' ? 5 : '',
          paddingBottom: category === 'animal' ? 5 : '',
          borderRadius: category === 'animal' ? 5 : '',
          color: category === 'animal' ? 'white' : 'black',
          cursor: 'pointer'
        }}
        >Animal</p>
      <p
        onClick={() => setCategory('flower', '5')}
        style={{
          backgroundColor: category === 'flower' ? 'red' : '',
          textAlign: 'center',
          paddingTop: category === 'flower' ? 5 : '',
          paddingBottom: category === 'flower' ? 5 : '',
          borderRadius: category === 'flower' ? 5 : '',
          color: category === 'flower' ? 'white' : 'black',
          cursor: 'pointer'
        }}
      >Flower</p>
      <p
        onClick={() => setCategory('christmas', '6')}
        style={{
          backgroundColor: category === 'christmas' ? 'red' : '',
          textAlign: 'center',
          paddingTop: category === 'christmas' ? 5 : '',
          paddingBottom: category === 'christmas' ? 5 : '',
          borderRadius: category === 'christmas' ? 5 : '',
          color: category === 'christmas' ? 'white' : 'black',
          cursor: 'pointer'
        }}
        >Christmas</p>
      <p
        onClick={() => setCategory('val', '7')}
        style={{
          backgroundColor: category === 'val' ? 'red' : '',
          textAlign: 'center',
          paddingTop: category === 'val' ? 5 : '',
          paddingBottom: category === 'val' ? 5 : '',
          borderRadius: category === 'val' ? 5 : '',
          color: category === 'val' ? 'white' : 'black',
          cursor: 'pointer'
        }}
      >Valentine's</p>
    </div>
  )
}

export default Category;
