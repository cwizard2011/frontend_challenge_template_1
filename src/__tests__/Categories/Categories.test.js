import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Categories from '../../Categories';

describe('<Categories />', () => {
  const wrapper = mount(<Categories />);
  it('should render without crashing', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});