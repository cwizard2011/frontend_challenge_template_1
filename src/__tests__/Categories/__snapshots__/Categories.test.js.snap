// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categories /> should render without crashing 1`] = `
<Category>
  <div
    style={
      Object {
        "backgroundColor": "#FAFAFA",
        "borderRadius": 5,
        "boxShadow": "0.5rem 0.5rem 3rem rgba(0,0,0,0.2)",
        "marginLeft": 10,
        "marginTop": 30,
        "paddingBottom": 10,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 20,
        "position": "absolute",
      }
    }
  >
    <h4>
      Categories
    </h4>
    <p
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "",
          "borderRadius": "",
          "color": "black",
          "cursor": "pointer",
          "paddingBottom": "",
          "paddingTop": "",
          "textAlign": "center",
        }
      }
    >
      French
    </p>
    <p
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "",
          "borderRadius": "",
          "color": "black",
          "cursor": "pointer",
          "paddingBottom": "",
          "paddingTop": "",
          "textAlign": "center",
        }
      }
    >
      Italian
    </p>
    <p
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "",
          "borderRadius": "",
          "color": "black",
          "cursor": "pointer",
          "paddingBottom": "",
          "paddingTop": "",
          "textAlign": "center",
        }
      }
    >
      Irish
    </p>
    <p
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "",
          "borderRadius": "",
          "color": "black",
          "cursor": "pointer",
          "paddingBottom": "",
          "paddingTop": "",
          "textAlign": "center",
        }
      }
    >
      Animal
    </p>
    <p
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "",
          "borderRadius": "",
          "color": "black",
          "cursor": "pointer",
          "paddingBottom": "",
          "paddingTop": "",
          "textAlign": "center",
        }
      }
    >
      Flower
    </p>
    <p
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "",
          "borderRadius": "",
          "color": "black",
          "cursor": "pointer",
          "paddingBottom": "",
          "paddingTop": "",
          "textAlign": "center",
        }
      }
    >
      Christmas
    </p>
    <p
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "",
          "borderRadius": "",
          "color": "black",
          "cursor": "pointer",
          "paddingBottom": "",
          "paddingTop": "",
          "textAlign": "center",
        }
      }
    >
      Valentine's
    </p>
  </div>
</Category>
`;
